### PR TITLE
[Bug Fix] Fixed the discrepacy with time using command #time and in quests.

### LINKED ISSUE #3700: Discrepancies in time
When setting time using `#time` or `#set time`, the time in game looks right, but the message is wrong.
image

Then, through quest, `$zonehour` is yet another our different.
image

Also noticed in `gm_commands\set\time.cpp` it is using `Strings::ZoneTime(zone->zone_time.getEQTimeZoneHr(), zone->zone_time.getEQTimeZoneHr())` in 3 locations
That last element should be `zone->zone_time.getEQTimeZoneMin())`

### PR DESCRIPTION
**Pull request by regneq**
Fixes #3700

image

### CODE REVIEW

File: zone/gm_commands/set/time.cpp [hunk 1]
```diff
@@ -9,33 +9,34 @@ void SetTime(Client *c, const Seperator *sep)
...
 
 	if (hours > 24) {
 		hours = 24;
 	}
 
 	uint8 real_hours = (
-		(hours - 1) > 0 ?
-		(hours - 1) :
+		(hours) > 0 ?
+		(hours) :
```
Comment (Kinglykrab) [inline zone/gm_commands/set/time.cpp:39]: Can remove `()` from this I reckon. :)

**Comment by Kinglykrab** [comment]
Can remove anything that's commented out if you want.

File: zone/gm_commands/set/time.cpp (hunk 1 above)
Comment (regneq) [reply zone/gm_commands/set/time.cpp:39]: okay  I'll do that.

File: zone/gm_commands/set/time.cpp [hunk 2]
```diff
@@ -47,21 +41,20 @@ void SetTime(Client *c, const Seperator *sep)
...
+			Strings::ZoneTime(hours, minutes)
 		).c_str()
 	);
 
-	zone->SetTime(real_hours, minutes);
+	zone->SetTime(real_hours + 1, minutes);
 
 	LogInfo(
-		"{} :: Setting world time to {} (Timezone: {})",
+		"{} :: Setting world time to {})",
```
Comment (Kinglykrab) [inline zone/gm_commands/set/time.cpp:56]: Typo?

File: zone/gm_commands/set/time.cpp [hunk 3]
```diff
@@ -47,21 +41,20 @@ void SetTime(Client *c, const Seperator *sep)
...
 	}
 
+
 	c->Message(
 		Chat::White,
 		fmt::format(
-			"Setting world time to {} (Timezone: {}).",
-			Strings::ZoneTime(hours, minutes),
-			Strings::ZoneTime(zone->zone_time.getEQTimeZoneHr(), zone->zone_time.getEQTimeZoneHr())
+			"Setting world time to {}).",
```
Comment (Kinglykrab) [inline zone/gm_commands/set/time.cpp:48]: Typo?

Comment (Kinglykrab) [reply zone/gm_commands/set/time.cpp:48]: `{}).`

File: zone/gm_commands/set/time.cpp (hunk 2 above)
Comment (Kinglykrab) [reply zone/gm_commands/set/time.cpp:56]: `{})`

Comment (regneq) [reply zone/gm_commands/set/time.cpp:56]: oops

**Review by joligario — APPROVED** [review]
So crazy we have to adjust time internally...

**Review by Kinglykrab — APPROVED** [review]

**Comment by regneq** [comment]
timescreen

here's the test I have done locally.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**